docs/software_requirements/queues.sdoc

```diff
@@ -0,0 +1,216 @@
+COMPONENT: Queues
+TITLE: Queue append item to the end
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to append a data item to the end of a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-4
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue append item to the start
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to append a data item in the first place of a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-5
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue remove item
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to remove a data item from a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-6
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue get item
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to get a data item from a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-7
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue get item timeout
+STATEMENT: >>>
+When a timout for the get mechanism to get a data item is configured, the zephyr RTOS shall wait
+that period to optain the data item from the queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-8
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue empty check
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to check if a queue is empty.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-9
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue look up last item
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to look up the last data item in the queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-10
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue look up first item
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to look up the first data item in the queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-11
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue insert
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to insert a data item behind another specific data item in the queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-12
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue append complete list
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to append a complete list of data items to the end of a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-13
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue add list
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to a complete list of data items to a queue.
```

Comment:
I don't understand this requirement. To me it seems a word is missing here.

Reply:
There was something missing indeed. Rephrased the requirement and the one which belongs to this one.